libs/viewer/src/SimpleViewer.cpp

```diff
@@ -783,6 +797,177 @@ void SimpleViewer::changeAllVisibility(utils::Entity entity, bool changeToVisibl
     }
 };
 
+static const char* formatToName[] = {
+    "R8",
+    "R8_SNORM",
+    "R8UI",
+    "R8I",
+    "STENCIL8",
+    "R16F",
+    "R16UI",
+    "R16I",
+    "RG8",
+    "RG8_SNORM",
+    "RG8UI",
+    "RG8I",
+    "RGB565",
+    "RGB9_E5",
+    "RGB5_A1",
+    "RGBA4",
+    "DEPTH16",
+    "RGB8",
+    "SRGB8",
+    "RGB8_SNORM",
+    "RGB8UI",
+    "RGB8I",
+    "DEPTH24",
+    "R32F",
+    "R32UI",
+    "R32I",
+    "RG16F",
+    "RG16UI",
+    "RG16I",
+    "R11F_G11F_B10F",
+    "RGBA8",
+    "SRGB8_A8",
+    "RGBA8_SNORM",
+    "UNUSED",
+    "RGB10_A2",
+    "RGBA8UI",
+    "RGBA8I",
+    "DEPTH32F",
+    "DEPTH24_STENCIL8",
+    "DEPTH32F_STENCIL8",
+    "RGB16F",
+    "RGB16UI",
+    "RGB16I",
+    "RG32F",
+    "RG32UI",
+    "RG32I",
+    "RGBA16F",
+    "RGBA16UI",
+    "RGBA16I",
+    "RGB32F",
+    "RGB32UI",
+    "RGB32I",
+    "RGBA32F",
+    "RGBA32UI",
+    "RGBA32I",
+    "EAC_R11",
+    "EAC_R11_SIGNED",
+    "EAC_RG11",
+    "EAC_RG11_SIGNED",
+    "ETC2_RGB8",
+    "ETC2_SRGB8",
+    "ETC2_RGB8_A1",
+    "ETC2_SRGB8_A1",
+    "ETC2_EAC_RGBA8",
+    "ETC2_EAC_SRGBA8",
+    "DXT1_RGB",
+    "DXT1_RGBA",
+    "DXT3_RGBA",
+    "DXT5_RGBA",
+    "DXT1_SRGB",
+    "DXT1_SRGBA",
+    "DXT3_SRGBA",
+    "DXT5_SRGBA",
+    "RGBA_ASTC_4x4",
+    "RGBA_ASTC_5x4",
+    "RGBA_ASTC_5x5",
+    "RGBA_ASTC_6x5",
+    "RGBA_ASTC_6x6",
+    "RGBA_ASTC_8x5",
+    "RGBA_ASTC_8x6",
+    "RGBA_ASTC_8x8",
+    "RGBA_ASTC_10x5",
+    "RGBA_ASTC_10x6",
+    "RGBA_ASTC_10x8",
+    "RGBA_ASTC_10x10",
+    "RGBA_ASTC_12x10",
+    "RGBA_ASTC_12x12",
+    "SRGB8_ALPHA8_ASTC_4x4",
+    "SRGB8_ALPHA8_ASTC_5x4",
+    "SRGB8_ALPHA8_ASTC_5x5",
+    "SRGB8_ALPHA8_ASTC_6x5",
+    "SRGB8_ALPHA8_ASTC_6x6",
+    "SRGB8_ALPHA8_ASTC_8x5",
+    "SRGB8_ALPHA8_ASTC_8x6",
+    "SRGB8_ALPHA8_ASTC_8x8",
+    "SRGB8_ALPHA8_ASTC_10x5",
+    "SRGB8_ALPHA8_ASTC_10x6",
+    "SRGB8_ALPHA8_ASTC_10x8",
+    "SRGB8_ALPHA8_ASTC_10x10",
+    "SRGB8_ALPHA8_ASTC_12x10",
+    "SRGB8_ALPHA8_ASTC_12x12",
+};
+
+
+std::string SimpleViewer::validateTweaks(const TweakableMaterial& tweaks) {
+    auto verifyTextured = [&]<typename T, bool MayContainFile, bool IsColor, bool IsDerivable>(std::string& result, const std::string& prompt, const TweakableProperty<T, MayContainFile, IsColor, IsDerivable>& prop, filament::Texture::InternalFormat expectedFormat = filament::Texture::InternalFormat::UNUSED) {
+        if (prop.isFile) {
+            if (prop.filename.getFileExtension() != "png") {
+                result += "ERROR: " + prompt + " is not a png.\n";
+            }
+
+            auto textureEntry = mTextures.find(prop.filename.asString());
+            if (textureEntry != mTextures.end()) {
```

Comment:
could invert condition here too

Reply:
Fixed in https://github.com/shapr3d/filament/pull/20/commits/ddf94a75580b2a0573fd9d18dc5bc78c44e25463